pkg/epp/datalayer/collector.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package datalayer
+
+import (
+	"context"
+	"sync"
+	"time"
+)
+
+// TODO:
+// currently the data store is expected to manage the state of multiple
+// Collectors (e.g., using sync.Map mapping pod to its Collector). Alternatively,
+// this can be encapsulated in this file, providing the data store with an interface
+// to only update on endpoint addition/change and deletion. This can also be used
+// to centrally track statistics such errors, active routines, etc.
+
+// Ticker implements a time source for periodic invocation.
+// Defined as an interface to allow mocking in tests.
+type Ticker interface {
+	Channel() <-chan time.Time
+	Stop()
+}
+
+// TimeTicker implements a Ticker based on time.Ticker.
+type TimeTicker struct {
+	*time.Ticker
+}
+
+// NewTimeTicker returns a new time.Ticker with the configured duration.
+func NewTimeTicker(d time.Duration) Ticker {
+	return &TimeTicker{
+		Ticker: time.NewTicker(d),
+	}
+}
+
+// Channel exposes the ticker's channel.
+func (t *TimeTicker) Channel() <-chan time.Time {
+	return t.C
+
+}
+
+// Collector runs the data collection for a single endpoint.
+type Collector struct {
+	// per-endpoint context and cancellation
+	ticker Ticker
```

Comment:
if you agree that no need for ticker interface, then we don't need ticker field in the struct nor as function arg.
we can create ticker inside the start loop.  
example here:
https://github.com/kubernetes-sigs/gateway-api-inference-extension/blob/main/pkg/epp/backend/metrics/pod_metrics.go#L93

Reply:
the caller (e.g., datastore should own the ticker interval (or else provide it as input so we can create the timer internally).
If created internally, it *should* to be stored on the object so it's `Stop()` method can be called (this is not strictly required from Go 1.23 onwards as GC can collect timers who are no longer referenced). An explicit `Stop()` might still be preferred (e.g., to avoid ticks after `Stop` has been called?).